### PR TITLE
chore(entities): clean up SubtypeTable and rowToElggStar

### PR DESCRIPTION
**Pull request by mrclay**
(in preparation for later work)

### CODE REVIEW

File: engine/classes/Elgg/Database/EntityTable.php [hunk 1]
```diff
@@ -76,66 +76,55 @@ function rowToElggStar($row) {
...
-		$new_entity = false;
-	
 		// Create a memcache cache if we can
 		static $newentity_cache;
 		if ((!$newentity_cache) && (is_memcache_available())) {
 			$newentity_cache = new \ElggMemcache('new_entity_cache');
 		}
 		if ($newentity_cache) {
-			$new_entity = $newentity_cache->load($row->guid);
+			$cached = $newentity_cache->load($row->guid);
```
Comment (hypeJunction) [inline engine/classes/Elgg/Database/EntityTable.php]: If we are going to allow updating subtype class dynamically,  do we want to return cached entity without checking instance?

Comment (hypeJunction) [inline engine/classes/Elgg/Database/EntityTable.php]: I am not sure what the lifespan of the cache is in this case. Does it propagate across requests?

**Comment by hypeJunction** [comment]
LGTM

File: engine/classes/Elgg/Database/EntityTable.php (hunk 1 above)
Comment (mrclay) [inline engine/classes/Elgg/Database/EntityTable.php]: Good point. This is a bug!

File: engine/classes/Elgg/Database/SubtypeTable.php [hunk 2]
```diff
@@ -317,12 +255,46 @@ function update($type, $subtype, $class = '') {
...
+	 * Empty the cache. Also invalidates the boot cache and memcache
+	 *
+	 * @return void
+	 */
+	private function invalidateCache() {
+		$this->cache = null;
+		_elgg_services()->boot->invalidateCache();
+		if (is_memcache_available()) {
+			(new \ElggMemcache('new_entity_cache'))->clear();
+		}
```
Comment (mrclay) [inline engine/classes/Elgg/Database/SubtypeTable.php]: @hypeJunction we just clear the memcache. That way we don't have to jump through the subtype code before returning a memcache entity.

**Pull request merged**